modules/attachment/variables.tf

```diff
@@ -1,6 +1,9 @@
-variable "profile_id" {
-  type        = string
-  description = "ID of the profile you wish to use for the attachment."
+variable "profile" {
```

Comment:
Was there a reason why you went with object? I think its might be easier to use (in Projects anyway) if we go with `profile_name` and `profile_version`. There will only ever be one name and one version for an attachment anyway